packages/cli/src/commands/help.ts

```diff
@@ -0,0 +1,212 @@
+import chalk from 'chalk';
+import stripAnsi from 'strip-ansi';
+import { LOGO, NAME } from '@vercel-internals/constants';
+
+const INDENT = ' '.repeat(2);
+const NEWLINE = '\n';
+const MAX_LINE_LENGTH = process.stdout.columns;
+
+export interface CommandOption {
+  name: string;
+  shorthand: string | null;
+  type: 'boolean' | 'string';
+  argument?: string;
+  deprecated: boolean;
+  description: string;
+  multi: boolean;
+}
+export interface CommandArgument {
+  name: string;
+  required: boolean;
+}
+export interface CommandExample {
+  name: string;
+  value: string | string[];
+}
+export interface Command {
+  name: string;
+  description: string;
+  arguments: CommandArgument[];
+  options: CommandOption[];
+  examples: CommandExample[];
+}
+
+function calcLineLength(line: string[]) {
+  return stripAnsi(lineToString(line)).length;
+}
+
+// Insert spaces in between non-whitespace items only
+function lineToString(line: string[]) {
+  let string = '';
+  for (let i = 0; i < line.length; i++) {
+    if (i === line.length - 1) {
+      string += line[i];
+    } else {
+      const curr = line[i];
+      const next = line[i + 1];
+      string += curr;
+      if (curr.trim() !== '' && next.trim() !== '') {
+        string += ' ';
+      }
+    }
+  }
+  return string;
+}
+
+function outputArrayToString(outputArray: string[]) {
+  return outputArray.reduce((acc, line) => acc + `${line}${NEWLINE}`, '');
+}
+
+/**
+ * Example: `▲ vercel deploy [path] [options]`
+ * @param command
+ * @returns
+ */
+function buildCommandSynopsisLine(command: Command) {
+  const line: string[] = [LOGO, chalk.bold(NAME), chalk.bold(command.name)];
+  if (command.arguments.length > 0) {
+    for (const argument of command.arguments) {
+      line.push(argument.required ? argument.name : `[${argument.name}]`);
+    }
+  }
+  if (command.options.length > 0) {
+    line.push('[options]');
+  }
+  return lineToString(line);
+}
+
+function buildCommandOptionLines(command: Command) {
+  // Filter out deprecated options.
+  command.options = command.options.filter(option => !option.deprecated);
```

Comment:
<!-- Suggestion comments propose improvements to the code. Mention specifically what should be changed and what the benefit would be.  -->
**suggestion (soft-blocking):** can we allow for intentionally undocumented as well?

```suggestion
  // Filter out deprecated and intentionally undocumented options
  command.options = command.options.filter(option => !option.deprecated && !option.description);
```